### PR TITLE
Make case classes for targets with ids.

### PR DESCRIPTION
**Pull request by toddburnside**

### CODE REVIEW

File: model/shared/src/main/scala/explore/model/TargetWithId.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    for {
+      id     <- c.get[Target.Id]("id")
+      target <- c.as[Target]
+    } yield TargetWithId(id, target)
+  )
+
+  implicit val eqTargetWithId: Eq[TargetWithId] = Eq.by(x => (x.id, x.target))
+}
+
+case class TargetWithOptId(optId: Option[Target.Id], target: Target)
```
Comment (toddburnside) [inline model/shared/src/main/scala/explore/model/TargetWithId.scala:61]: Would it be preferable to put all the case classes at the top of the file and the objects further down?

**Review by cquiroz — APPROVED** [review]
Great, looks better this way

File: model/shared/src/main/scala/explore/model/TargetWithId.scala (hunk 1 above)
Comment (rpiaggio) [reply model/shared/src/main/scala/explore/model/TargetWithId.scala:61]: I think I'd prefer that yes

**Review by rpiaggio — APPROVED** [review]
LGTM. It improves readability a lot. Thank you!

**Pull request merged**